### PR TITLE
Chronicle Redis Persistence

### PR DESCRIPTION
**Pull request by nick-cromwell**
Note, this is dependent on PR #30 in the Chronicle project [https://github.com/snatch-dev/Chronicle/pull/30]

### CODE REVIEW

File: src/Chronicle.Integrations.Redis/Chronicle.Integrations.Redis.csproj [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (GooRiOn) [inline src/Chronicle.Integrations.Redis/Chronicle.Integrations.Redis.csproj]: could you migrate to .NET Standard 2.1? I've already updated base package

File: src/Chronicle.Integrations.Redis/persistence/RedisSagaLog.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        public RedisSagaLog(IDistributedCache cache)
+        {
+            this.cache = cache;
+        }
+
+        public async Task<IEnumerable<ISagaLogData>> ReadAsync(SagaId id, Type sagaType)
+        {
+            if (String.IsNullOrWhiteSpace(id))
+            {
+                throw new ArgumentException(nameof(id));
```
Comment (GooRiOn) [inline src/Chronicle.Integrations.Redis/persistence/RedisSagaLog.cs]: please use `ChronicleException`  instead

File: src/Chronicle.Integrations.Redis/persistence/RedisSagaLog.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            }
+            if (sagaType is null)
+            {
+                throw new ArgumentException(nameof(sagaType));
+            }
+
+            List<RedisSagaLogData> sagaLogDatas = new List<RedisSagaLogData>();
```
Comment (GooRiOn) [inline src/Chronicle.Integrations.Redis/persistence/RedisSagaLog.cs]: use var instead

File: src/Chronicle.Integrations.Redis/persistence/RedisSagaLog.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            return deserializedSagaLogDatas;
+        }
+
+        public async Task WriteAsync(ISagaLogData logData)
+        {
+            if (logData is null)
+            {
+                throw new ArgumentException(nameof(logData));
+            }
+            IList<ISagaLogData> sagaLogDatas = (await ReadAsync(logData.SagaId, logData.SagaType)).ToList();
```
Comment (GooRiOn) [inline src/Chronicle.Integrations.Redis/persistence/RedisSagaLog.cs]: var

File: src/Chronicle.Integrations.Redis/persistence/RedisSagaLog.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+            var serializedSagaLogDatas = JsonConvert.SerializeObject(sagaLogDatas);
+            await cache.SetStringAsync(LogId(logData.SagaId, logData.SagaType), serializedSagaLogDatas);
+        }
+
+        public async Task DeleteAsync(SagaId sagaId, Type sagaType)
+        {
+            if (String.IsNullOrWhiteSpace(sagaId))
+            {
+                throw new ArgumentException(nameof(sagaId));
```
Comment (GooRiOn) [inline src/Chronicle.Integrations.Redis/persistence/RedisSagaLog.cs]: `ChronicleException`

Comment (nick-cromwell) [reply src/Chronicle.Integrations.Redis/persistence/RedisSagaLog.cs]: All ArgumentExceptions are now ChronicleExceptions

File: src/Chronicle.Integrations.Redis/Extensions.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+
+namespace Chronicle.Integrations.Redis
+{
+    public static class Extensions
+    {
+        private static string DeserializationError => "Could not deserialize given appsettings.";
+
+        public static IChronicleBuilder UseMongoPersistence(this IChronicleBuilder builder, string appSettingsSection, IConfiguration configuration)
```
[the remark this answers is not in the document]
Comment (nick-cromwell) [reply src/Chronicle.Integrations.Redis/Extensions.cs]: Haha oops...

**Comment by nick-cromwell** [comment]
Resolved the issues. Just a reminder this is dependent on changes to SagaId in the main repo. Let me know if there's anything else

**Pull request merged**